variation-commons-core/src/main/java/uk/ac/ebi/eva/commons/core/models/AbstractVariant.java

```diff
@@ -181,6 +181,13 @@ public String getMainId() {
         return mainId;
     }
 
+    public void renormalizeVariant(long newStart, long newEnd, String newReference, String newAlternate) {
+        this.start = newStart;
+        this.end = newEnd;
+        this.reference = newReference;
+        this.alternate = newAlternate;
```

Comment:
These values shouldn't be assigned without any previous checks, for instance, `start` and `end` can't be negative numbers. Could you please make these assignments based on (maybe private) setters with appropriate checks, and add some tests?